crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -291,8 +291,13 @@ def disable_non_core_proposals
 
   def prepare_nodes_for_os_upgrade
     upgrade_nodes = NodeObject.all.reject { |node| node.admin? || node[:platform] == "windows" }
+    admin_node = NodeObject.admin_node
+    admin_node["target_platform"] = admin_node["provisioner"]["default_os"]
+    admin_node.save
```

Comment:
@vuntz You mean just for the admin node? Or for all nodes?

I guess we could do it also on restore. Might a bit more complicated to implement it there though.